cluster/src/main/java/org/apache/iotdb/cluster/query/LocalQueryExecutor.java

```diff
@@ -602,16 +602,15 @@ public long querySingleSeriesByTimestamp(SingleSeriesQueryRequest request)
     }
   }
 
-  public ByteBuffer getAllMeasurementSchema(ByteBuffer planBuffer)
+  public ByteBuffer getAllMeasurementSchema(MeasurementSchemaRequest request)
       throws CheckConsistencyException, IOException, MetadataException {
     dataGroupMember.syncLeaderWithConsistencyCheck(false);
 
-    ShowTimeSeriesPlan plan = (ShowTimeSeriesPlan) PhysicalPlan.Factory.create(planBuffer);
+    ShowTimeSeriesPlan plan = (ShowTimeSeriesPlan) PhysicalPlan.Factory.create(request.planBinary);
     List<ShowTimeSeriesResult> allTimeseriesSchema;
-    allTimeseriesSchema =
-        getCMManager()
-            .showLocalTimeseries(
-                plan, new QueryContext(SessionManager.getInstance().requestQueryId(false)));
+    RemoteQueryContext queryContext =
+        queryManager.getQueryContext(request.getRequester(), request.getQueryId());
+    allTimeseriesSchema = getCMManager().showLocalTimeseries(plan, queryContext);
```

Comment:
Looks reasonable, but if we pass the queryId this way, we need to send an endQuery request from the coordinator node after the query completes to clean up the resources in the queryManager, otherwise there will be a memory leak. So far I haven't found any code related to resource cleaning, what do you think?

Reply:
@OneSizeFitsQuorum I debug this case and see that the queryId is assigned at QueryTask.call(). (Here the queryId is 31)
![image](https://user-images.githubusercontent.com/3821212/154840951-c5510949-3bbf-4fc6-8abd-429166d63b5f.png)

And the query resource is released in `FetchResultTask` and `SessionManager.closeDataSet()` (The query id is 31 as well)

![image](https://user-images.githubusercontent.com/3821212/154841036-2b5c9a9a-fa1e-4ab4-b314-e5bc6499f9cf.png)
![image](https://user-images.githubusercontent.com/3821212/154841094-e742ee0b-d3a8-4a5b-a8be-837d846e3c7a.png)

For any query, it will execute the query firstly and get a query id. Then it will fetch results, if necessary. Fetching results request will be sent with this query id. When fetching results ends, it will release all local and remote query resources.

So I think there's no resource leak. And you can reference `ClusterSessionManager.releaseQueryResource()` to get the logics of remote query resource release.

Reply:
`ClusterSessionManager.releaseQueryResource()` will release resource in the remote node only when the remote node is registered by `RemoteQueryContext.registerRemoteNode(Node node, RaftNode header)`. 

Now it's possible that showTimeseriesPlan holds the state of the remote node, but it doesn't seem to register the remote node during execution?

<img width="1049" alt="image" src="https://user-images.githubusercontent.com/32640567/154854773-c6a82275-3e2d-4452-bce7-df5c24f07f43.png">

Reply:
@OneSizeFitsQuorum Nice catch! It's my mistake to miss the `registerRemoteNode()`. I add this in a finally block in `private void showRemoteTimeseries(PartitionGroup group, ShowTimeSeriesPlan plan, QueryContext context, ShowTimeSeriesHandler handler)` in `CMManager`.